### PR TITLE
mypy inspired fixes / updates

### PR DESCRIPTION
**Pull request by ThomasWaldmann**

### CODE REVIEW

File: src/borg/crypto/key.py [hunk 1]
```diff
@@ -417,6 +417,7 @@ def init_ciphers(self, manifest_data=None):
 
 class FlexiKey:
     FILE_ID = "BORG_KEY"
+    STORAGE = KeyBlobStorage.NO_STORAGE  # override in subclass
```
Comment (RonnyPfannschmidt) [inline src/borg/crypto/key.py:420]:
```suggestion
    STORAGE: ClassVar[KeyBlobStorage] = KeyBlobStorage.NO_STORAGE  # override in subclass
```
for extra clarity

File: src/borg/upgrade.py [hunk 2]
```diff
@@ -129,7 +129,7 @@ def upgrade_zlib_and_level(meta, data):
             borg1_header_fmt = Struct(">I")
```
Comment (RonnyPfannschmidt) [inline src/borg/upgrade.py:129]: i think this should be extracted so its only made once

File: src/borg/crypto/key.py (hunk 1 above)
Comment (ThomasWaldmann) [reply src/borg/crypto/key.py:420]: hmm, `KeyBlobStorage.NO_STORAGE` is a str.

Comment (RonnyPfannschmidt) [reply src/borg/crypto/key.py:420]: :facepalm: my mistake, usage looked like a enum

File: src/borg/upgrade.py (hunk 2 above)
Comment (ThomasWaldmann) [reply src/borg/upgrade.py]: done

File: src/borg/crypto/key.py (hunk 1 above)
Comment (ThomasWaldmann) [reply src/borg/crypto/key.py]: see 2705ccb

**Pull request merged**